src/ds/ds-device-common.h

```diff
@@ -0,0 +1,62 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "hw-monitor.h"
+#include "ds-private.h"
```

Comment:
Is it possible to keep ds-private included only on cpp files? (and forward declare on h files)
Only if possible (and can be considered for a next PR) I know it was implemented this way before.

Reply:
no, because ds namespace is not known without it